indexd/index/drivers/alchemy.py

```diff
@@ -1096,6 +1097,28 @@ def get(self, did, expand=True):
 
             return record.to_document_dict()
 
+    def get_with_nonstrict_prefix(self, did, expand=True):
+        """
+        Attempt to retrieve a record both with and without a prefix.
+        Proxies 'get' with provided id.
+        If not found but prefix matches default, attempt with prefix stripped.
+        If not found and id has no prefix, attempt with default prefix prepended.
+        """
+        try:
+            # Try to resolve id as provided
+            record = self.get(did, expand=expand)
+        except NoRecordFound as e:
+            DEFAULT_PREFIX = self.config.get("DEFAULT_PREFIX")
+            match = re.match(r"(dg\.[0-9a-f]{4}\/)(.+)", did)
+            if match and match.group(1) == DEFAULT_PREFIX:
+                record = self.get(match.group(2), expand=expand)
+            elif match is None:
```

Comment:
i think something like this could let us avoid the regex
```
if "/" in did: (because we use UUIDs, there's only a `/` if there's a prefix)
    if did.split("/")[0] == DEFAULT_PREFIX:
        try without prefix
else:
    try with prefix
```
and then of course we can have a test case with a GUID that exists, but a prefix different from DEFAULT_PREFIX, and make sure we return 404